### PR TITLE
Return an error if kubeseal release is not found

### PR DESCRIPTION
**Pull request by Waterdrips**
## Description

Based on suggestion in  #749 to return error on non-200 status code rather than download the 404 HTML page.

We now show the user we got a non 200 status code, and return.

Signed-off-by: Alistair Hey <alistair@heyal.co.uk>

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## How Has This Been Tested?

```
GOARCH=arm GOARM=6 go build

#On ARM 
pi@raspberrypi:~ $ ./faas-cli cloud seal --download
Starting download of kubeseal v0.9.6, this could take a few moments.
Could not find release, http status code was 404, release may not exist for this archetecture

# On AMD64
heyal@heyal-xps:~/go/src/github.com/openfaas/faas-cli$ ./faas-cli cloud seal --download
Starting download of kubeseal v0.9.6, this could take a few moments.
Download completed, please run:

  chmod +x /tmp/kubeseal
  /tmp/kubeseal --version
  sudo install /tmp/kubeseal /usr/local/bin/

```

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: commands/cloud.go [hunk 1]
```diff
@@ -228,9 +229,13 @@ func downloadBinary(client *http.Client, url, name string) (string, error) {
 	}
 
 	res, err := client.Do(req)
+
 	if err != nil {
 		return "", err
 	}
+	if res.StatusCode != 200 {
+		return "", errors.New(fmt.Sprintf("could not find release, http status code was %d, release may not exist for this archetecture", res.StatusCode))
```
Comment (alexellis) [inline commands/cloud.go]: Architecture

File: commands/cloud.go [hunk 2]
```diff
@@ -228,9 +229,13 @@ func downloadBinary(client *http.Client, url, name string) (string, error) {
 	}
 
 	res, err := client.Do(req)
+
 	if err != nil {
 		return "", err
 	}
+	if res.StatusCode != 200 {
```
Comment (alexellis) [inline commands/cloud.go]: Http.statusOK

**Comment by alexellis** [comment]
Please can you provide a body for the commit when you fix up the typo?

**Comment by Waterdrips** [comment]
> Please can you provide a body for the commit when you fix up the typo?

The requested changes and commit body have been pushed

**Pull request merged**